src/httpClient.ts

```diff
@@ -51,142 +47,58 @@ export class HttpClient {
   async fetch(
     input: string,
     init?: RequestInit | undefined,
-    maxRetries: number = DEFAULT_MAX_RETRIES,
-    retryDelay: number = DEFAULT_RETRY_DELAY,
   ): Promise<Response> {
     const url = this.baseURL + input;
 
-    let lastError: Error | null = null;
-    let currentRetryDelay = retryDelay;
-
-    for (let attempt = 0; attempt < maxRetries; attempt++) {
-      try {
-        const response = await this.attemptFetch(url, init);
-
-        // Check for OK response
-        if (response.ok) return response;
-
-        // If not ok, then generate an error message
-        const errorMessage = this.generateErrorMessage(response, url);
-        lastError = new Error(errorMessage);
-
-        // Fail fast for non-transient errors
-        if (this.nonTransientError(response)) break;
+    const finalInit = {
+      ...init,
+      headers: {
+        ...init?.headers,
+        "Content-Type": init?.headers?.["Content-Type"] ?? "application/json",
+        authorization: `Bearer ${this.apiKey}`,
+      },
+    };
 
-        // Retry logic for transient errors
-        if (this.shouldRetry(response)) {
-          await this.waitForRetry(response, currentRetryDelay);
-          currentRetryDelay *= 2;
-        } else {
-          // For other errors, throw an error
-          throw new Error(errorMessage);
-        }
-      } catch (error) {
-        lastError = error as Error;
+    const IS_DEBUG = this.debug;
```

Comment:
nit: naming: usually caps are used for package level constants, maybe rename to `isDebug`

Reply:
Fixed